### PR TITLE
resolved: Fix incorrect use of OpenSSL BUF_MEM

### PR DESCRIPTION
**Pull request by t8m**
Fixes: #12763

### CODE REVIEW

**Comment by yuwata** [comment]
I may not understand the change, but does this code support all 1.1.x and 1.0.x? Or, in other words, is it not necessary to conditionalize the code?

File: src/resolve/resolved-dnstls-openssl.c [hunk 1]
```diff
@@ -10,6 +10,7 @@
 #include "io-util.h"
 #include "resolved-dns-stream.h"
 #include "resolved-dnstls.h"
+#include "string-util.h"
```
Comment (yuwata) [inline src/resolve/resolved-dnstls-openssl.c]: Including `<string.h>` is enough here.

**Comment by t8m** [comment]
This code should work for both 1.1.x and 1.0.x.

**Comment by yuwata** [comment]
Thanks. LGTM now.

**Pull request merged**